youtube_dl/extractor/medaltv.py

```diff
@@ -0,0 +1,127 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+from ..compat import compat_str
+from ..utils import (
+    try_get,
+    float_or_none,
+    int_or_none
+)
+
+
+class MedalTVIE(InfoExtractor):
+    _VALID_URL = r'(?:https://)?(?:www\.)?medal\.tv(?:/clips)?/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://medal.tv/clips/34934644/3Is9zyGMoBMr',
+        'md5': '7b07b064331b1cf9e8e5c52a06ae68fa',
+        'info_dict': {
+            'id': '34934644',
+            'ext': 'mp4',
+            'title': 'Quad Cold',
+            'description': 'Medal,https://medal.tv/desktop/',
+            'uploader': 'MowgliSB',
+            'timestamp': 1603165266,
+            'upload_date': '20201020',
+            'uploader_id': 10619174,
+        }
+    }, {
+        'url': 'https://medal.tv/clips/36787208',
+        'md5': 'b6dc76b78195fff0b4f8bf4a33ec2148',
+        'info_dict': {
+            'id': '36787208',
+            'ext': 'mp4',
+            'title': 'u tk me i tk u bigger',
+            'description': 'Medal,https://medal.tv/desktop/',
+            'uploader': 'Mimicc',
+            'timestamp': 1605580939,
+            'upload_date': '20201117',
+            'uploader_id': 5156321,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        url = 'https://medal.tv/clips/{}'.format(video_id)
+        webpage = self._download_webpage(url, video_id)
+
+        hydration_data = self._search_regex(r'<script>.*hydrationData\s*=\s*({.+?})\s*</script>', webpage, 'hydration data', default='{}')
+        parsed = self._parse_json(hydration_data, video_id)
+
+        author_info = next(iter(parsed.get('profiles').values()))
+        clip_info = parsed.get('clips').get(video_id)
+
+        if('error' in clip_info):
+            error_code = clip_info.get('error')
+            if(error_code == 404):
+                raise ExtractorError('That clip does not exist.', expected=True, video_id=video_id)
+            else:
+                raise ExtractorError('An unknown error occurred ({}).'.format(error_code), video_id=video_id)
+
+        source_aspect_ratio = clip_info.get('sourceWidth') / clip_info.get('sourceHeight')
```

Comment:
same if one of the values is `None`.

Reply:
Resolved by using `try_get`:
```python
source_aspect_ratio = try_get(clip_info, lambda x: x.get('sourceWidth')/x.get('sourceHeight'), float) or (16/9)
```

Otherwise defaulting to standard aspect ratio 16:9